.github/workflows/cve-daily.yml

```diff
@@ -0,0 +1,373 @@
+      - name: Run CVE tests
+        run: |
+          echo "⚓️ 📤 Checking out candi/image_versions.yml from ${TAG}..."
+          git checkout --force "${TAG}" -- candi/image_versions.yml
+          echo "⚓️ 🏎 Running CVE tests on ${TAG}..."
+          make cve-base-images
+
+      # <template: unlink_bin_step>
+      - name: Unlink binary cache
+        id: unlink_bin_step
+        if: always()
+        run: |
+          rm bin
+      # </template: unlink_bin_step>
+    # </template: cve_tests>
+
+  test_cve_report_main:
+    name: Deckhouse images on main
+    needs:
+      - skip_tests_repos
+      - test_cve_base_images_main
+    env:
+      REPO: "dev-registry.deckhouse.io/sys/deckhouse-oss"
+      TAG: "main"
+    # <template: cve_tests>
+    runs-on: [self-hosted, regular]
+    steps:
+
+      # <template: checkout_full_step>
+      - name: Checkout sources
+        uses: actions/checkout@v2.4.0
+        with:
+          fetch-depth: 0
+      # </template: checkout_full_step>
+
+      # <template: login_dev_registry_step>
+      - name: Check dev registry credentials
+        id: check_dev_registry
+        env:
+          HOST: ${{secrets.DECKHOUSE_DEV_REGISTRY_HOST}}
+        run: |
+          if [[ -n $HOST ]]; then echo "::set-output name=has_credentials::true"; fi
+      - name: Login to dev registry
+        uses: docker/login-action@v1.10.0
+        if: ${{ steps.check_dev_registry.outputs.has_credentials == 'true' }}
+        with:
+          registry: ${{ secrets.DECKHOUSE_DEV_REGISTRY_HOST }}
+          username: ${{ secrets.DECKHOUSE_DEV_REGISTRY_USER }}
+          password: ${{ secrets.DECKHOUSE_DEV_REGISTRY_PASSWORD }}
+          logout: false
+      # </template: login_dev_registry_step>
+
+      # <template: login_readonly_registry_step>
+      - name: Check readonly registry credentials
+        id: check_readonly_registry
+        env:
+          HOST: ${{secrets.DECKHOUSE_REGISTRY_READ_HOST}}
+        run: |
+          if [[ -n $HOST ]]; then echo "::set-output name=has_credentials::true"; fi
+      - name: Login to readonly registry
+        uses: docker/login-action@v1.10.0
+        if: ${{ steps.check_readonly_registry.outputs.has_credentials == 'true' }}
+        with:
+          registry: ${{ secrets.DECKHOUSE_REGISTRY_READ_HOST }}
+          username: ${{ secrets.DECKHOUSE_REGISTRY_READ_USER }}
+          password: ${{ secrets.DECKHOUSE_REGISTRY_READ_PASSWORD }}
+          logout: false
+      # </template: login_readonly_registry_step>
+
+      # <template: link_bin_step>
+      - name: Link binary cache
+        id: link_bin_step
+        run: |
+          ln -s ~/deckhouse-bin-cache bin
+      # </template: link_bin_step>
+      - name: Set repo and tag for matrix run
+        if: matrix.tag != ''
+        run: |
+          REPO="registry.deckhouse.io/deckhouse/fe"
+          echo "REPO=${REPO}" >> $GITHUB_ENV
+          TAG=$(echo "${{ matrix.tag }}" | grep -Eo "^\S+")
```

Comment:
Using gha expressions (`${{ ... }}`) inside the shell script text is not reliable. It may corrupt the script text if expression result contains quotes, newlines or other fancy symbols. It is better pass them using env:

```
- name: ...
  env:
    matrix_tag: ${{ matrix.tag }}
  run: |
    echo "${matrix_tag}" | ...
```

For example, see [passing job output and a context value via envs](https://github.com/deckhouse/deckhouse/blob/main/.github/workflows/deploy-alpha.yml#L357-L362). Also see passing gha expressions results as [json-encoded strings](https://github.com/deckhouse/deckhouse/blob/main/.github/workflows/deploy-alpha.yml#L767-L770).

Reply:
I've applied your suggestion about GHA expressions in c2832026961731ceb2aa38eaadbefa25da5d18b3.

Reply:
I've applied your suggestion about GHA expressions in c2832026961731ceb2aa38eaadbefa25da5d18b3.